upstairs/src/lib.rs

```diff
@@ -1617,15 +1630,15 @@ impl Buffer {
         assert_eq!(offset % self.block_size, 0);
         assert_eq!(response.data.len(), self.block_size);
         if !response.block_contexts.is_empty() {
-            self.data[offset..][..response.data.len()]
+            self.block_mut(offset / self.block_size)
```

Comment:
style nit: cache `let block = offset / self.block_size;` similar to how it's done elsewhere?

Reply:
Also done in 0b24644d8678fcfcdac84d8f8d0cc9da9af0d146